extensions/import/src/dialogs/derivedColumnDialog.ts

```diff
@@ -0,0 +1,393 @@
+			}
+
+			this._transformationTable = view.modelBuilder.declarativeTable().withProps({
+				height: '100%',
+				columns: [
+					{
+						displayName: 'Specify Transformation',
+						valueType: azdata.DeclarativeDataType.component,
+						isReadOnly: false,
+						width: '200px',
+						headerCssStyles: headerLeft,
+						rowCssStyles: styleLeft
+					}
+				],
+				CSSStyles: {
+					'table-layout': 'fixed'
+				},
+				dataValues: transformationTableData
+			}).component();
+
+
+
+
+			this._applyButton.onClick(async e => {
+				const requiredColNames = [];
+				const numCols = this._transformationTable.columns.length - 1;
+				for (let index = 0; index < numCols; index++) {
+					requiredColNames[index] = this._transformationTable.columns[index].displayName;
+				}
+				const transExamples = [];
+				const transExampleIndices = [];
+
+				for (let index = 0; index < this._transformationTable.dataValues.length; index++) {
+					const example = (<azdata.InputBoxComponent>this._transformationTable.dataValues[index][numCols].value).value as string;
+					if (example === '') {
+						continue;
+					}
+					transExamples.push(example);
+					transExampleIndices.push(index);
+				}
+				if (transExamples.length > 0) {
+					const response = await this._provider.sendLearnTransformationRequest({
+						columnNames: requiredColNames,
+						transformationExamples: transExamples,
+						transformationExampleRowIndices: transExampleIndices
+					});
+					this.currentTransformation = response.transformationPreview;
+					for (let index = 0; index < this.currentTransformation.length; index++) {
+						(<azdata.InputBoxComponent>this._transformationTable.dataValues[index][this._transformationTable.columns.length - 1].value).placeHolder = this.currentTransformation[index];
+					}
+					this.clearAndAddTransformationContainerComponents(true);
+				}
+				this.enableDoneButton();
+			});
+
+			const specifyDerivedColNameTable = view.modelBuilder.declarativeTable().withProps({
+				columns: [
+					{
+						displayName: 'Column Name',
+						valueType: azdata.DeclarativeDataType.string,
+						isReadOnly: false,
+						width: '150px',
+						headerCssStyles: headerLeft,
+						rowCssStyles: styleLeft,
+					}
+				],
+				height: '100vh',
+				CSSStyles: {
+
+				}
+			}).component();
+
+			specifyDerivedColNameTable.onDataChanged(e => {
+				this.currentDerivedColumnName = specifyDerivedColNameTable.dataValues[0][0].value as string;
+				this.enableDoneButton();
+			});
+
+			const specifyDerivedColNameTableData: azdata.DeclarativeTableCellValue[][] = [];
+			const colNameTableRow: azdata.DeclarativeTableCellValue[] = [];
+			colNameTableRow.push({
+				value: ''
+			});
+			specifyDerivedColNameTableData.push(colNameTableRow);
+			specifyDerivedColNameTable.dataValues = specifyDerivedColNameTableData;
+
+			this._transformationContainer = view.modelBuilder.flexContainer().withLayout({
+				flexFlow: 'column',
+				width: '700px',
```

Comment:
it'd be great if you used `em` instead since `px` is static while `em` is relative based on screen size